pkg/scripts/install.go

```diff
@@ -13,177 +15,234 @@ import (
 	"github.com/openshift/source-to-image/pkg/util"
 )
 
-// Installer interface is responsible for installing scripts needed to run the build
+// Installer interface is responsible for installing scripts needed to run the
+// build.
 type Installer interface {
 	InstallRequired(scripts []string, dstDir string) ([]api.InstallResult, error)
 	InstallOptional(scripts []string, dstDir string) []api.InstallResult
 }
 
-// NewInstaller returns a new instance of the default Installer implementation
-func NewInstaller(image string, scriptsURL string, docker docker.Docker, auth dockerClient.AuthConfiguration) Installer {
-	return &installer{
-		image:      image,
-		scriptsURL: scriptsURL,
-		docker:     docker,
-		downloader: NewDownloader(),
-		pullAuth:   auth,
-		fs:         util.NewFileSystem(),
-	}
+// ScriptHandler provides an interface for various scripts source handlers.
+type ScriptHandler interface {
+	Get(script string) *api.InstallResult
+	Install(*api.InstallResult) error
+	SetDestinationDir(string)
 }
 
-type installer struct {
-	image      string
-	scriptsURL string
-	docker     docker.Docker
-	pullAuth   dockerClient.AuthConfiguration
-	downloader Downloader
-	fs         util.FileSystem
+// UrlScriptHandler handles script download using URL.
+type UrlScriptHandler struct {
```

Comment:
We discussed this with @soltysh on IRC. We keep this exported as it implements a public interface and allow upstream to use this package and create their own script handlers (and register them).